### DISCUSSION TITLE
How to receive messages ?

### DISCUSSION THREAD

**Discussion by tianalemesle** [AsamK/signal-cli#804] [Q&A]
Hello,
I'm sorry to ask a such dumb question, but I can't seem to find any documentation on this.
How can I make `signal-cli` listen to new messages ?
For example, when sending a message in *Note to Self* from Signal for Android, I instantly see it on Signal Desktop for Linux.
So I would like to begin with that.
Thanks

**Comment by exquo** [comment]
Try

	signal-cli daemon

or

	signal-cli receive --timeout=-1

See the [man page](https://github.com/AsamK/signal-cli/blob/master/man/signal-cli.1.adoc)

**Reply by KaKi87** [reply]
Thank you for your answer.

`signal-cli -o json daemon` is nice.

Also thank you for your work on [exquo/signal-libs-build](https://github.com/exquo/signal-libs-build).

Can you also tell me how to find info on currently linked devices and how to link multiple devices ?

Thanks !

**Reply by calipee** [reply]
Did you check the man page?

You should be able to do it like described [here](https://github.com/AsamK/signal-cli/blob/master/man/signal-cli.1.adoc#:~:text=listDevices,of%20linked%20devices) .

> signal-cli [--config CONFIG] [-h | -v | -a ACCOUNT | --dbus | --dbus-system] command [command-options]
> listDevices
> Show a list of linked devices.

**Reply by KaKi87** [reply]
I'm sorry, I didn't meant *devices* but *linked accounts*. I linked one account to my computer device using `signal-cli link`, and would like to list linked accounts and add other accounts. Thanks

PS: I wrongly said *devices* because I was thinking *smartphones*, as those accounts are created using phone numbers from the *smartphone* app.

**Reply by KaKi87** [reply]
I'm also wondering how to unlink an account from `signal-cli` rather than the smartphone. Thanks

**Reply by KaKi87** [reply]
@exquo, @calipee, please ? Thanks

**Reply by exquo** [reply]
You can only unlink from the master / primary device. If you are using signal on your smartphone - that's your master device for that phone number. If instead you have done `signal-cli register`, then signal-cli is master, and you can use `signal-cli removeDevice …` to remove devices you have previously linked with `signal-cli addDevice`. See [wiki](https://github.com/AsamK/signal-cli/wiki/Linking-other-devices-(Provisioning)).

**Reply by KaKi87** [reply]
Is there any way to remove `signal-cli`'s own access to a remote master device without using the device itself ?
For example, Signal Desktop has a *clear application data* feature.

And, how can I `link` multiple smartphone devices on a single computer device ?

Thanks

**Reply by exquo** [reply]
> Is there any way to remove signal-cli's own access to a remote master device without using the device itself ?

Not sure, try `signal-cli removeDevice …`. But it might work only if signal-cli is master. 

> And, how can I link multiple smartphone devices on a single computer device ?

Not possible. Signal on a smartphone can only be master, and only one device can be master.

**Reply by KaKi87** [reply]
In that case, how does Signal Desktop allow logout, and how can multiple users of the same computer link their respective Signal accounts using `signal-cli` ?

Thanks
